src/utils/password-hashing.ts

```diff
@@ -0,0 +1,24 @@
+import bcrypt, { genSaltSync } from 'bcryptjs';
+import { SERVER_ERROR_MESSAGES } from '@/constants/server-error-messages';
+
+const { ENV_KEYS_MISSING, COMPLETION_ERROR } = SERVER_ERROR_MESSAGES;
+
+/**
+ * Hashes a password using bcrypt
+ *
+ * @param password - The password to hash
+ * @returns A promise that resolves with the hashed password
+ */
+export const hashPassword = async (password: string) => {
+  const salt = process.env.PASSWORD_SALT_ROUNDS;
+
+  if (!salt) {
+    throw new Error(ENV_KEYS_MISSING(['PASSWORD_SALT_ROUNDS']));
+  }
+
+  if (Number.isNaN(Number(salt))) {
+    throw new Error(COMPLETION_ERROR('hashing the password'));
+  }
+
+  return bcrypt.hash(password, genSaltSync(Number(salt)));
```

Comment:
I appreciate your review and suggestion.

I researched the difference between `genSalt` and `genSaltSync` and realized your suggestion would be a performance improvement in this case because it's an asynchronous function. I've changed the method.